src/HDF5.jl

```diff
@@ -2369,11 +2368,11 @@ function h5lt_dtype_to_text(dtype_id)
     status = ccall((:H5LTdtype_to_text, libhdf5_hl), Herr, (Hid, Ptr{UInt8}, Int, Ref{Csize_t}), dtype_id, C_NULL, 0, len)
     status < 0 && error("Error getting dtype text representation")
 
-    buf = Vector{UInt8}(undef, len[])
+    buf = Base.StringVector(len[]-1)
```

Comment:
cc @kleinhenz  
I took another look at this and realized it's much better to allocate string buffers as follows. Reduces an allocation and this is more idiomatic Julia